cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/FastMap.scala

```diff
@@ -0,0 +1,428 @@
+package com.twitter.cassovary.util.collections
+
+import java.util
+
+import it.unimi.dsi.fastutil.ints._
+import it.unimi.dsi.fastutil.objects._
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.{specialized => sp}
+
+/**
+ * Specialized, fast and memory efficient implementation of mutable map.
+ *
+ * Provides basic operations only. See [[FastMapWrapper]] for how to use
+ * it as a [[mutable.Map]].
+ *
+ * Use constructor provided in the companion object.
+ */
+trait FastMap[@sp(Int, Long, AnyRef) A, @sp(Int, Long, AnyRef) B] {
+
+  /**
+   * Adds key-value pair to the map.
+   */
+  def +=(key: A, value: B): this.type
+
+  /**
+   * Removes key-value pair with a given key.
+   */
+  def -=(key: A): this.type
+
+  /**
+   * Get is optimized for retrieving elements contained in the Map.
+   *
+   * If the element is not in the map, returns `0` of given type (or false, null).
+   */
+  def get(key: A): B
```

Comment:
If we wanted to return Option, we need to create our own specialized... This is possible, but I prefered to have implementation as fast as possible... 
There is also a problem with fastutil specialized maps and Option: they return predefined value (defaults to 0) on get, when key is not in the map. This is confusing in general setting, because we cannot distinguish it from the value that could be a value connected to the key, so to return None, we would have to call contains first...

We have similar problem with += that accepts a Tuple (in Scala), but we need to implement one ourselves to have it specialized...